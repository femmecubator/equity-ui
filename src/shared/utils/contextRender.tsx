import { render, RenderResult } from '@testing-library/react';

import { EquityThemeProvider } from '../../theme';

const contextRender = (ui: React.ReactNode): RenderResult => {
  return render(<EquityThemeProvider>{ui}</EquityThemeProvider>);
};

export default contextRender;
