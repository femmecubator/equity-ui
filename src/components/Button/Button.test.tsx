import { describe, expect, it } from 'vitest';
import { screen } from '@testing-library/react';
import { Button } from '.';
import contextRender from '../../shared/test/contextRender';

describe('Button component', () => {
  it('Button should render correctly', () => {
    contextRender(<Button>Hello!</Button>);
    const button = screen.getByRole('button');
    expect(button).toBeInTheDocument();
  });
});
