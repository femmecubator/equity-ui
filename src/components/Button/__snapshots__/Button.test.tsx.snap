// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > matches snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: relative;
  border: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  gap: 6px;
  box-sizing: border-box;
  outline: none;
  cursor: pointer;
  -webkit-transition: all 0.15s ease;
  transition: all 0.15s ease;
  padding: 6px 12px;
  border-radius: 900px;
  background-color: #761fb1;
  color: #ffffff;
  border: 1px solid #761fb1;
}

.emotion-0:hover:not(:disabled) {
  background-color: #a82dfd;
  border-color: #a82dfd;
}

.emotion-0:focus:not(:disabled) {
  background-color: #761fb1;
  border: 3px solid #113264;
}

.emotion-0:disabled {
  background-color: #e0e0e0;
  color: #828282;
  border-color: #9c9c9c;
  cursor: not-allowed;
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
}

.emotion-2 {
  font-size: 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: 'Inter',-apple-system,BlinkMacSystemFont,'Segoe UI',system-ui,sans-serif;
  font-weight: 600;
  font-style: normal;
  font-stretch: normal;
  letter-spacing: 0;
  line-height: 20px;
  paragraph-indent: 0;
  paragraph-spacing: 0;
  text-case: none;
  color: #222222;
  margin: 0;
  color: inherit;
}

<button
    class="emotion-0 emotion-1"
    color="purple"
    shape="pill"
    type="button"
  >
    <p
      class="emotion-2 emotion-3"
      size="2"
    >
      Hello!
    </p>
  </button>
</DocumentFragment>
`;

exports[`Button component > matches snapshot with all props 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: relative;
  border: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  gap: 6px;
  box-sizing: border-box;
  outline: none;
  cursor: pointer;
  -webkit-transition: all 0.15s ease;
  transition: all 0.15s ease;
  padding: 6px 12px;
  border-radius: 900px;
  background-color: #761fb1;
  color: #ffffff;
  border: 1px solid #761fb1;
}

.emotion-0:hover:not(:disabled) {
  background-color: #a82dfd;
  border-color: #a82dfd;
}

.emotion-0:focus:not(:disabled) {
  background-color: #761fb1;
  border: 3px solid #113264;
}

.emotion-0:disabled {
  background-color: #e0e0e0;
  color: #828282;
  border-color: #9c9c9c;
  cursor: not-allowed;
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
}

.emotion-2 {
  color: inherit;
}

.emotion-3 {
  font-size: 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: 'Inter',-apple-system,BlinkMacSystemFont,'Segoe UI',system-ui,sans-serif;
  font-weight: 600;
  font-style: normal;
  font-stretch: normal;
  letter-spacing: 0;
  line-height: 20px;
  paragraph-indent: 0;
  paragraph-spacing: 0;
  text-case: none;
  color: #222222;
  margin: 0;
  color: inherit;
}

<button
    class="emotion-0 emotion-1"
    color="purple"
    shape="pill"
    type="button"
  >
    <svg
      class="emotion-2"
      color="#222222"
      height="24"
      style="display: block; vertical-align: middle;"
      width="24"
    >
      <use
        href="/src/icons/sprite.svg#statistics"
      />
    </svg>
    <p
      class="emotion-3 emotion-4"
      size="2"
    >
      Full Props Button
    </p>
    <svg
      class="emotion-2"
      color="#222222"
      height="24"
      style="display: block; vertical-align: middle;"
      width="24"
    >
      <use
        href="/src/icons/sprite.svg#complete"
      />
    </svg>
  </button>
</DocumentFragment>
`;
